### DISCUSSION TITLE
Do we need...Guidance?

### DISCUSSION THREAD

**Discussion by St33lMouse** [oobabooga/text-generation-webui#2154] [Ideas]
https://github.com/microsoft/guidance

Guidance enables you to control modern language models more effectively and efficiently than traditional prompting or chaining. Guidance programs allow you to interleave generation, prompting, and logical control into a single continuous flow matching how the language model actually processes the text.

**Comment by miles-du** [comment]
Looks interesting. ...And this could be used in ooba? Like as an extension?

**Reply by St33lMouse** [reply]
Someone would have to incorporate Guidance into Ooba in some way. I have no idea how to do it :)

**Reply by NEFlame** [reply]
https://github.com/Maximilian-Winter/guidance

Someone already made a fork of it with llama-cpp-python support.

**Reply by St33lMouse** [reply]
Didn't know that. Thanks!
